x-pack/test/plugin_api_integration/test_suites/encrypted_saved_objects/encrypted_saved_objects_api.ts

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import { SavedObject } from 'src/legacy/server/saved_objects/service';
+import { KibanaFunctionalTestDefaultProviders } from '../../../types/providers';
+
+// eslint-disable-next-line import/no-default-export
+export default function({ getService }: KibanaFunctionalTestDefaultProviders) {
+  const es = getService('es');
+  const chance = getService('chance');
+  const supertest = getService('supertest');
+
+  const SAVED_OBJECT_WITH_SECRET_TYPE = 'saved-object-with-secret';
+
+  async function getRawSavedObjectAttributes(id: string) {
+    const {
+      _source: { [SAVED_OBJECT_WITH_SECRET_TYPE]: savedObject },
+    } = await es.get({
+      id: `${SAVED_OBJECT_WITH_SECRET_TYPE}:${id}`,
+      type: '_doc',
+      index: '.kibana',
+    });
+
+    return savedObject;
+  }
+
+  describe('within a default space', () => {
```

Comment:
__note:__ working on a space-aware test suite (it can go in a follow up though, but we'll have it one way or another anyway).